### PR TITLE
[Ready] Add API Call & Example OPs

### PR DESCRIPTION
**Pull request by drcege**
### Summary of Changes
- Added `APIModel`
  - Compatible with `OpenAI` and `DashScope` requests.
- New Operations Utilizing API Calls
  - `CalibrateQAMapper`
  - `CalibrateQueryMapper`
  - `CalibrateResponseMapper`
- Related Updates
  - Unit tests passed locally.
  - Added docs.

### TODO
- [ ] ~Implement API request retry~ (to be refined in a separate PR)

### Note
Recommended to merge after #454, handle code conflicts carefully, and update the operator count correctly.

### CODE REVIEW

File: configs/config_all.yaml [hunk 1]
```diff
@@ -53,6 +53,18 @@ hpo_config: null                                            # path to a configur
 process:
   # Mapper ops. Most of these ops need no arguments.
   - audio_ffmpeg_wrapped_mapper:                            # simple wrapper for FFmpeg audio filters
+  - calibrate_qa_mapper:                                    # calibrate question-answer pairs based on reference text.
+      api_url:                                                # API URL. Defaults to DJ_API_URL environment variable.
```
Comment (BeachWang) [inline configs/config_all.yaml:57]: 漏掉api_model参数

File: data_juicer/ops/mapper/calibrate_qa_mapper.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        messages = [{
+            'role': 'system',
+            'content': self.system_prompt
+        }, {
+            'role': 'user',
+            'content': self.build_input(sample)
+        }]
+        output = client(messages, **self.api_params)
+
+        parsed_q, parsed_a = self.parse_output(output)
```
Comment (BeachWang) [inline data_juicer/ops/mapper/calibrate_qa_mapper.py:103]: API call经常出现异常，会返回None，这里没做判断是否会出错？

File: data_juicer/ops/mapper/calibrate_qa_mapper.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        client = get_model(self.model_key, rank=rank)
+
+        messages = [{
+            'role': 'system',
+            'content': self.system_prompt
+        }, {
+            'role': 'user',
+            'content': self.build_input(sample)
+        }]
+        output = client(messages, **self.api_params)
```
Comment (BeachWang) [inline data_juicer/ops/mapper/calibrate_qa_mapper.py:101]: 建议需要加个try_num参数，当返回None或者不能解析时重试。

File: data_juicer/ops/mapper/calibrate_qa_mapper.py (hunk 2 above)
Comment (drcege) [reply data_juicer/ops/mapper/calibrate_qa_mapper.py:103]: 改为空字符串了

File: configs/config_all.yaml (hunk 1 above)
Comment (drcege) [reply configs/config_all.yaml:57]: 加了

**Review by BeachWang — APPROVED** [review]
LGTM

**Pull request merged**